### PR TITLE
Add GitHub actions workflow

### PR DESCRIPTION
**Pull request by moh60**
1. Add build steps to lint and compile the extension

### CODE REVIEW

**Comment by idillon-sfl** [comment]
looks good to me

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: vscode-bitbake CI/CD
```
Comment (deribaucourt) [inline .github/workflows/main.yml:1]: There is a merge commit in your PR that I think should be eliminated with a rebase on staging.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Change Directory
+      run: cd path/to/your/project
```
Comment (deribaucourt) [inline .github/workflows/main.yml]: This template is not adapted yet to our project. I suggest squashing it directly with the next commit.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -23,42 +23,14 @@ jobs:
       with:
         node-version: ${{ matrix.node-version }}
 
-    - name: Change Directory
-      run: cd path/to/your/project
+    - name: Clean install
```
Comment (deribaucourt) [inline .github/workflows/main.yml:26]: To respect the commit title convention, please prefix your message with the kind of action you are addressing: Fix/Chore/Refactor/Feature. I guess "Feature: Add GitHub actions workflow"

**Review by deribaucourt — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (moh60) [reply .github/workflows/main.yml:26]: done

File: .github/workflows/main.yml (hunk 2 above)
Comment (moh60) [reply .github/workflows/main.yml]: done

**Review by WilsonZiweiWang — APPROVED** [review]

**Review by deribaucourt — APPROVED** [review]

**Pull request merged**